oak-search/src/main/java/org/apache/jackrabbit/oak/plugins/index/search/spi/editor/FulltextDocumentMaker.java

```diff
@@ -187,7 +189,10 @@ public D makeDocument(NodeState state, boolean isUpdate, List<PropertyState> pro
         }
 
         if (indexingRule.isFulltextEnabled()) {
-            indexFulltextValue(document, name);
+            boolean shouldAdd = propertyRegex == null || propertyRegex.matcher(name).find();
```

Comment:
I would move the `propertyRegex` declaration inside the condition where it is used
```suggestion
            Pattern propertyRegex = definition.getPropertyRegex();
            boolean shouldAdd = propertyRegex == null || propertyRegex.matcher(name).find();
```